<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

## 1 Learning to interrupt

In [3]:
while True:
    print('*', end='')

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

## 2 Restarting the Kernal

In [4]:
print = 'PRINT'
print ('Hello World!')

TypeError: 'str' object is not callable

In [77]:
print ('Hello World!')

Hello World!


## 3 Timing Stuff

In [17]:
squares = [i**2 for i in range (10)]
print (squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [20]:
squares = list()
for i in range (10):
    squares.append(i**2)
print (squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


- Both codes give the same results

In [23]:
%%timeit -n 100000 -r 7
squares = [i**2 for i in range (10)]

218 ns ± 103 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)


In [76]:
%%timeit -n 100000 -r 7
for i in range (10):
    squares.append(i**2)

223 ns ± 103 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)


- to find out how much time does it take for the cell to execute the command, the system took a total of 7 runs, each run consisting of 100000 loops (ie. run the cell 100000 times, and repeat it for a total of 7 times)
- 223 ns indicates refers to the mean time to run the cell is 223 nano-seconds
- 103 ns indicates refers to the standard deviation of the spread of the timings collected